zkvm/src/constraints.rs

```diff
@@ -119,35 +119,45 @@ impl Constraint {
                 Ok((r1cs::LinearCombination::from(o), assignment))
             }
             Constraint::Not(c1) => {
+                // Compute the input linear combination and its secret assignment
                 let (x_lc, x_assg) = c1.flatten(cs)?;
+
+                // Compute assignments for all the wires
                 let (xy_assg, xw_assg, y_assg) = match x_assg {
-                    Some(x_assg) => {
-                        let is_zero = x_assg.ct_eq(&Scalar::zero());
-                        let y_assg = Scalar::conditional_select(
+                    Some(x) => {
+                        let is_zero = x.ct_eq(&Scalar::zero());
+                        let y = Scalar::conditional_select(
                             &Scalar::zero(),
                             &Scalar::one(),
                             is_zero.into(),
                         );
-                        let w_assg =
-                            Scalar::conditional_select(&x_assg, &Scalar::one(), is_zero.into());
-                        let w_assg = w_assg.invert();
-                        (Some((x_assg, y_assg)), Some((x_assg, w_assg)), Some(y_assg))
+                        let w = Scalar::conditional_select(&x, &Scalar::one(), is_zero.into());
+                        let w = w.invert();
+                        (Some((x, w)), Some((x, w)), Some(y))
```

Comment:
i introduced a bug here (`w` instead of `y` in the first tuple), oops (fixed in a later commit).